src/contracts/middleware/BLSPubkeyRegistry.sol

```diff
@@ -0,0 +1,208 @@
+        _;
+    }
+
+    constructor(
+        IRegistryCoordinator _registryCoordinator,
+        IBLSPublicKeyCompendium _pubkeyCompendium
+    ){
+        registryCoordinator = _registryCoordinator;
+        pubkeyCompendium = _pubkeyCompendium;
+    }
+
+    /**
+     * @notice Registers the `operator`'s pubkey for the specified `quorumNumbers`.
+     * @param operator The address of the operator to register.
+     * @param quorumNumbers The quorum numbers the operator is registering for, where each byte is an 8 bit integer quorumNumber.
+     * @param pubkey The operator's BLS public key.
+     * @dev access restricted to the RegistryCoordinator
+     * @dev Preconditions:
+     *         1) `quorumNumbers` has no duplicates
+     *         2) `quorumNumbers.length` != 0
+     *         3) `quorumNumbers` is ordered in ascending order
+     *         4) the operator is not already registered
+     */
+    function registerOperator(address operator, bytes memory quorumNumbers, BN254.G1Point memory pubkey) external onlyRegistryCoordinator returns(bytes32){
+        //calculate hash of the operator's pubkey
+        bytes32 pubkeyHash = BN254.hashG1Point(pubkey);
+
+        require(pubkeyHash != ZERO_PK_HASH, "BLSPubkeyRegistry.registerOperator: cannot register zero pubkey");
+        //ensure that the operator owns their public key by referencing the BLSPubkeyCompendium
+        require(pubkeyCompendium.pubkeyHashToOperator(pubkeyHash) == operator,"BLSPubkeyRegistry.registerOperator: operator does not own pubkey");
+        // update each quorum's aggregate pubkey
+        _processQuorumApkUpdate(quorumNumbers, pubkey);
+        // update the global aggregate pubkey
+        _processGlobalApkUpdate(pubkey);
+        // emit event so offchain actors can update their state
+        emit PubkeyAdded(operator, pubkey);
+        return pubkeyHash;
+    }
+
+    /**
+     * @notice Deregisters the `operator`'s pubkey for the specified `quorumNumbers`.
+     * @param operator The address of the operator to deregister.
+     * @param quorumNumbers The quourm numbers the operator is deregistering from, where each byte is an 8 bit integer quorumNumber.
+     * @param pubkey The public key of the operator.
+     * @dev access restricted to the RegistryCoordinator
+     * @dev Preconditions:
+     *         1) `quorumNumbers` has no duplicates
+     *         2) `quorumNumbers.length` != 0
+     *         3) `quorumNumbers` is ordered in ascending order
+     *         4) the operator is not already deregistered
+     *         5) `quorumNumbers` is the same as the parameter use when registering
+     *         6) `pubkey` is the same as the parameter used when registering
+     */   
+    function deregisterOperator(address operator, bytes memory quorumNumbers, BN254.G1Point memory pubkey) external onlyRegistryCoordinator returns(bytes32){
+        bytes32 pubkeyHash = BN254.hashG1Point(pubkey);
+
```

Comment:
am I crazy or is the same check (as compared to the `registerOperator` function) that the operator owns the pubkey missing here?
Really just trying to understand why the `registerOperator` and `deregisterOperator` functions do different checks and whether this is a potential issue or not

Reply:
I believe the preconditions for the deregister function ensure that the operator provided pubkey will be the same as the one that was registered with.

Reply:
what @Sidu28 is detailing is what i intended as well. would you prefer we did the check here as well? how do you feel about this @ChaoticWalrus ?

Reply:
I think having the check here as well would be safer. IMO a slightly increased gas cost is worth it for the peace of mind here (also makes it easier to review).